### PR TITLE
DEVOPS-27143 - remove references to instancelabel and fragmentkey

### PR DESCRIPTION
**Pull request by EvertonCalgarotto**
InstanceLabel is a feature that enabled the use of an existing RDS cluster to create another database under it. This is no longer used as we are spinning up a new cluster by application.
So we are removing the support of it.

### CODE REVIEW

**Review by leandrorichardtoledo — APPROVED** [review]

File: config/postgres-exporter/deployment.yaml [hunk 1]
```diff
@@ -86,11 +86,16 @@ spec:
               containerPort: 9187
               protocol: TCP
           livenessProbe:
-            initialDelaySeconds: 0
+            exec:
+              command:
+              - sh
+              - -c
+              - wget -q -O - http://localhost:9187/metrics | grep '^pg_up ' | awk '{if ($2 == 1) exit 0; else exit 1;}'
```
Comment (drewwells) [inline config/postgres-exporter/deployment.yaml:93]: What's the idea here? Typically, we just want to know the metrics pod is up

File: docs/index.md [hunk 2]
```diff
@@ -407,7 +387,7 @@ DatabaseClaim:
 
     * Status: 
       - DbCreatedAt: Time the database was created.
-      - MatchedLabel: The name of the label that was successfully matched against the fragment key names in the db-controller configMap
+      - MatchedLabel: The name of the label that was successfully matched in the db-controller configMap
```
Comment (drewwells) [inline docs/index.md:390]: Is this gone now?

File: go.mod [hunk 3]
```diff
@@ -6,7 +6,6 @@ toolchain go1.22.4
 
 require (
 	github.com/Masterminds/sprig v2.22.0+incompatible
-	github.com/armon/go-radix v1.0.0
```
Comment (drewwells) [inline go.mod:9]: 🥳

**Review by drewwells — APPROVED** [review]

File: config/postgres-exporter/deployment.yaml (hunk 1 above)
Comment (EvertonCalgarotto) [reply config/postgres-exporter/deployment.yaml:93]: this is related to another ticket (https://infoblox.atlassian.net/browse/DEVOPS-27128), to restart the pod when there's an error in reading the credentials.

**Pull request merged**